lib/app/features/gallery/views/components/albums/album_thumbnail.dart

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/gallery/providers/gallery_provider.c.dart';
+import 'package:ion/app/features/gallery/views/components/shimmer_loading_cell.dart';
+import 'package:photo_manager/photo_manager.dart';
+import 'package:photo_manager_image_provider/photo_manager_image_provider.dart';
+
+class AlbumThumbnail extends ConsumerWidget {
+  const AlbumThumbnail({required this.albumId, super.key});
+
+  final String albumId;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final previewAsync = ref.watch(albumPreviewProvider(albumId));
+
+    return previewAsync.maybeWhen(
+      data: (asset) {
+        if (asset == null) {
+          return ClipRRect(
```

Comment:
we may create a private widget to use also in `orElse` , wdyt?

Reply:
done